src/components/Popups/TransactionPopup.tsx

```diff
@@ -17,25 +19,41 @@ export default function TransactionPopup({
   hash,
   success,
   summary,
+  swapProtocol,
 }: {
   hash: string
   success?: boolean
   summary?: string
+  swapProtocol?: SwapProtocol
 }) {
   const { chainId } = useActiveWeb3React()
 
   const theme = useTheme()
 
+  const isGnosisProtocolHash = swapProtocol === SwapProtocol.COW
+
+  const explorer = 'View on ' + (isGnosisProtocolHash ? 'Gnosis Protocol Explorer' : 'block explorer')
+
+  const body = chainId && (
```

Comment:
I'm a bit confused with "will be defined" and the hook returns `... | undefined` used in the same sentence.
This will create a ts warning that would need to be suppressed. Is that what we want to do?